server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveIndexAction.java

```diff
@@ -528,22 +528,33 @@ protected void doExecute(Task task, Request request, final ActionListener<Respon
         static void resolveIndices(
             String[] names,
             IndicesOptions indicesOptions,
-            Metadata metadata,
-            IndexAbstractionResolver resolver,
+            ClusterState clusterState,
+            IndexNameExpressionResolver resolver,
             List<ResolvedIndex> indices,
             List<ResolvedAlias> aliases,
-            List<ResolvedDataStream> dataStreams,
-            boolean includeDataStreams
+            List<ResolvedDataStream> dataStreams
         ) {
-            List<String> resolvedIndexAbstractions = resolver.resolveIndexAbstractions(names, indicesOptions, metadata, includeDataStreams);
-            SortedMap<String, IndexAbstraction> lookup = metadata.getIndicesLookup();
+            // redundant check to ensure that we don't resolve the list of empty names to "all" in this context
+            if (names.length == 0) {
+                return;
+            }
```

Comment:
It's great to make this check explicit. Can we also move the check of `if (localIndices != null)` from the caller inside this method? I think it makes sense for checks to be done in one place.

Reply:
pushed [2fee7b2](https://github.com/elastic/elasticsearch/pull/92820/commits/2fee7b2b7ba7c4f90220fc7c4ef72cbca2e4abca)